testing/e2e/upgradeGinkgo_test.go

```diff
@@ -0,0 +1,69 @@
+package e2e_ginkgo
+
+import (
+	"time"
+
+	tools "github.com/elastic/elastic-agent/testing/e2e/tools"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("Smoketests", func() {
+
+	Describe("Elastic Agent Install and Upgrade", Ordered, func() {
+		var enrollmentToken tools.EnrollmentAPIKey
+
+		// Setup: executed once before all specs within this Describe block
+		BeforeAll(func() {
+			By("Downloading elastic agent")
+			Expect(tools.DownloadElasticAgent(agentVersion)).To(Succeed())
+		})
+		// Setup: executed before each spec withing this Describe block
+		// I.e. we Create a new policy and token before each spec
+		BeforeEach(func() {
+			By("Create policy")
+			policy, err := client.CreatePolicy()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Create enrollment token")
+			enrollmentToken, err = client.CreateEnrollmentAPIKey(policy)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		// Spec: The test case
+		It("is online after upgrade", func() {
+			By("Installing & enrolling EA")
+			// gexec.Start() returns a session object that can be used to interact with the process
+			session, err := tools.EnrollElasticAgent(clusterConfig.FleetConfig.Url, enrollmentToken.APIKey, agentVersion)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(session, 2*time.Minute).Should(gexec.Exit(0))
+
+			By("Wait for agent to be healthy(online)")
+			Eventually(client.GetAgentStatus).WithTimeout(2 * time.Minute).WithPolling(5 * time.Second).Should(BeEquivalentTo("online"))
+
+			By("Upgrade elastic agent")
+			Expect(client.UpgradeAgent("8.6.1")).To(Succeed())
+
+			By("Wait for agent to be healthy(online)")
+			Eventually(client.GetAgentStatus).WithTimeout(5 * time.Minute).WithPolling(5 * time.Second).Should(BeEquivalentTo("online"))
```

Comment:
Let's replace Gingko with the testify package: https://pkg.go.dev/github.com/stretchr/testify#section-readme

There are similar replacements for most of the functions you are  using, like https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually

It will be more verbose but it is what we are used to using.